website/docs/basics/blocks-special-use-cases.md

```diff
@@ -86,3 +86,87 @@ With WordPress Block Editor hooks, we can fix this by not adding the selector to
 **Implementation**
 
 Here you can see our example of this hook used in the [column block](https://github.com/infinum/eightshift-frontend-libs/blob/develop/blocks/init/src/Blocks/custom/column/column-hooks.js).
+
+## Only use my custom blocks
+
+You can remove all core blocks and any other blocks provided by third-party plugins by adding this filter to your `src/Blocks/Blocks.php` class.
+
+Filter goes in the register method:
+```php
+// Limits the usage of only custom project blocks.
+add_filter('allowed_block_types', [ $this, 'getAllBlocksList' ], 10, 2);
+```
+
+The callback method is set in the parent `AbstractBlocks` class.
+
+## I want to use my custom blocks but add some core or third party plugins blocks
+
+You create a new callback method for the `allowed_block_types` filter.
+
+Filter goes in the register method:
+```php
+// Limits the usage of only custom project blocks.
+add_filter('allowed_block_types', [ $this, 'allowedBlocks' ], 10, 2);
+```
+
+Custom callback method:
+```php
+/**
+  * Filter what blocks are displayed in the block editor.
+  *
+  * @param array|bool $allowedBlockTypes Array of block type slugs, or boolean to enable/disable all.
+  * @param object     $post The post resource data.
+  *
+  * @return array
+  */
+public function allowedBlocks($allowedBlockTypes, object $post): array
+{
```

Comment:
`$allowedBlockTypes` can be `boolean` or `array` so there should be a check before doing any merging

```
if (is_array($allowedBlockTypes)) {
	$allowedBlocks = array_merge(parent::getAllBlocksList($allowedBlockTypes, $post), $allowedBlockTypes);
} else {
	$allowedBlocks = parent::getAllBlocksList($allowedBlockTypes, $post);
}
// additional logic here.

// Remove duplicates.
return array_keys(array_flip($allowedBlocks));
``` 

I got burned by this on one occasion.